### ISSUE TITLE
Graphics issues

### ISSUE THREAD

**Issue by rforberger** [ivan19871002/Thinkpad-P15V-Gen1-Hackintosh#9]
Hi,

I have tried your OC config, but I am having graphics issues when MacOS wants to switch to graphics mode when booting.

I get a blank screen.

I have a Thinkpad P15 with an Intel Corporation CometLake-H GT2 [UHD Graphics] (rev 05) iGPU.

I am using Hybrid Graphics setting in the BIOS.

I have tried to uncomment certain settings in the DeviceProperties, without success.

Any ideas what could be wrong?

Best regards,
rforberger

**Comment by ivan19871002** [comment]
https://dortania.github.io/GPU-Buyers-Guide/#so-if-my-gpu-is-natively-supported-why-do-i-need-lilu-and-whatevergreen

**Comment by ivan19871002** [comment]
https://dortania.github.io/GPU-Buyers-Guide/modern-gpus/intel-gpu.html#native-intel-igpus

**Comment by rforberger** [comment]
You mean, you have another iGPU than I?

**Comment by ivan19871002** [comment]
I think you need read whatevergreen  doc

**Comment by rforberger** [comment]
The only platform id that works is 3EA50009.
I have added options, but they don't seem to have effect:

```
 <key>AAPL,ig-platform-id</key>
                                <data>PqUACQ==</data>
                                <key>AAPL,slot-name</key>
                                <string>Internal@0,2,0</string>
                                <key>AAPL,GfxYTile</key>
                                <data>AQAAAA==</data>
                                <key>enable-hdmi-dividers-fix</key>
                                <data>AQAAAA==</data>
                                <key>dpcd-max-link-rate</key>
                                <data>FAAAAA==</data>
                                <key>enable-dpcd-max-link-rate-fix</key>
                                <data>AQAAAA==</data>
                                <key>enable-lspcon-support</key>
                                <data>AQAAAA==</data>
                                <key>enable-max-pixel-clock-override</key>
                                <data>AQAAAA==</data>
                                <key>enable-hdmi20</key>
                                <data>AQAAAA==</data>
                                <key>framebuffer-patch-enable</key>
                                <data>AQAAAA==</data>
                                <key>framebuffer-fbmem</key>
                                <data>AACQAA==</data>
                                <key>framebuffer-stolenmem</key>
                                <data>AAAwAQ==</data>
                                <key>framebuffer-unifiedmem</key>
                                <data>AAAAgA==</data>
                                <key>framebuffer-con1-enable</key>
                                <data>AQAAAA==</data>
                                <key>framebuffer-con0-enable</key>
                                <data>AQAAAA==</data>
                                <key>framebuffer-con0-has-lspcon</key>
                                <data>AQAAAA==</data>
                                <key>framebuffer-con0-preferred-lspcon-mode</key>
                                <data>AAAAAA==</data>
                                <key>framebuffer-con2-enable</key>
                                <data>AQAAAA==</data>
                                <key>framebuffer-con2-has-lspcon</key>
                                <data>AQAAAA==</data>
                                <key>framebuffer-con2-preferred-lspcon-mode</key>
                                <data>AAAAAA==</data>
                                <key>hda-gfx</key>
                                <string>onboard-1</string>
```

**Comment by rforberger** [comment]
I mixed up Bigendian with Littleendian. So my system booted with a wrong Platform ID, but now using correct Platform IDs, I get no graphics.

Platform IDs tried: 09009B3E, 06009B3E, 00009B3E, 0000923E, 0000003E, 0000A53E, 0400A53E, 0500A53E, 0900A53E, 0500A63E

Any ideas?

**Comment by rforberger** [comment]
I got the 3d accelaration running with platform ID 0500A53E.

But I am still having problems with HDMI output:

It's totally unclear to me, how to model the connector patches.

For the conX number, should I use con0,con1,con2 or con1,con2,con3? Your solution uses con1,con2,con3 here.
For the index of the connector, should I use 0,1,2 or 1,2,3 here? Your solution uses 1,2,3 here, but Hackintool on my PC says the connectors have indexes 0,1,2!
For the busid, Hackintool shows me 0x00 (LVDS), 0x05 (DP) and 0x04 (DP). Your solution though uses 0x05,0x06,0x04!
For the pipe, Hackintool shows 18 on all connectors on my PC, so I used 0x1200. Your solution uses 0x0900 here!
For the connector type I don't know if I should use 20 00 00 00 for the LVDS and 00 04 00 00 for the two DP or if I should 00 08 00 00 for one DP. Your solution uses 00 04 00 00, 00 04 00 00, 00 08 00 00 here!
As the flags I am using 0xC7030000 as suggested in a forum for awaking from sleep.

So any ideas how I should patch my connectors? I have tried a couple of permutations, but always black screen!
